test/AWS.Deploy.CLI.IntegrationTests/Helpers/TestEnvironmentVariableManager.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using AWS.Deploy.Orchestration.Utilities;
+
+namespace AWS.Deploy.CLI.IntegrationTests.Helpers
+{
+    public class TestEnvironmentVariableManager : IEnvironmentVariableManager
+    {
+        public readonly Dictionary<string, string> store = new Dictionary<string, string>();
+
+        public string GetEnvironmentVariable(string variable)
+        {
+            return store.ContainsKey(variable) ? store[variable] : null;
+        }
+
+        public void SetEnvironmentVariable(string variable, string value)
+        {
+            if (string.Equals(variable, "AWS_DOTNET_DEPLOYTOOL_WORKSPACE"))
+                store[variable] = value;
+            else
+                Environment.SetEnvironmentVariable(variable, value);
```

Comment:
Why is `AWS_DOTNET_DEPLOYTOOL_WORKSPACE` a special case? If this is a test implementation of IEnvironmentVariableManager, let's not impact the actual environment. So store all variables in the dictionary.

Reply:
There are certain environment variables (like `TEMP` and `TMP`) which we override if a custom workspace is set. These overrides need to recognized by different command line processes like `cdk deploy..`. That us why we set an explicit environment variable

Reply:
After a discussion, the EnvironmentVariableManager is not passed to the CDKProjectHandler. Please add a backlog item for us to do that work. For now, I'm fine with this change.